### ISSUE TITLE
Virtualized hosts failing to start guest VM after qemu update 6.0.0 -> 6.1.0

### ISSUE THREAD

**Issue by PhDyellow** [NixOS/nixpkgs#141596]
### Describe the bug
I am using `singularity-tools.buildImage` to create a portable singularity image. It was working until recently, but now it stops at:
```
iPXE (http://ipxe.org) 00:03.0 CA00 PCI2.10 PnP PMM+1FF8EFF0+1FEEEFF0 CA00



Booting from ROM...
Probing EDD (edd=off to disable)... ok
```
No progress is made form there, even when left overnight. 

### Steps To Reproduce

Run `nix-build singularity_image.nix -o ~/r-singularity-aus-bio.img -v`

where `singularity_image.nix` is:

```nix
{ pkgs
  ? import (builtins.fetchGit {
  name = "nixos-master";
  url = https://github.com/NixOS/nixpkgs/;
  ref = "refs/heads/master"; #using bleeding edge packages
  # rev = "e1fc1a80a071c90ab65fb6eafae5520579163783"; # good
  rev = "988da51d9c49a1d43fe28a5f92394f5dbc16eede"; #bad
})
    { #the attributes to import
    }
}:

let
  name = "r-singularity-aus-bio";
in
with pkgs; singularity-tools.buildImage { 
  name = name;
  }
```
### Expected behavior
The above steps should produce a singularity `sif` file at `~/r-singularity-aus-bio.img`.

`Probing EDD (edd=off to disable)... ok` should be followed by `cloading kernel modules` within a few seconds.

### Additional context
Using git bisect, I found that the problem was introduced in 988da51d, which is an update for `qemu` from 6.0.0 to 6.1.0. The problem seems to be in all commits to master after that point.

### Notify maintainers

qemu: @eelco @qyliss 
singularity-tools: @cfhammill @jbedo 
### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.71, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11pre322215.9bf75dd50b7"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: singularity-tools
# a list of nixos modules affected by the problem
module:
```

**Comment by PhDyellow** [comment]
Oracle Virtualbox 6.1.26, with a windows host.

**Comment by veehaitch** [comment]
We have the same issue when running a [basic nixos-test](https://github.com/yaxitech/ragenix/blob/882cc1cfd54bc060578d218fdd724d3bf890f99d/flake.nix#L213-L243) from a virtualized host, e.g., a GitHub Actions runner. It works fine for revisions before merging #135877.

**Comment by PhDyellow** [comment]
`-cpu host` was working for me as well.

**Comment by PhDyellow** [comment]
Perhaps 

```nix
self: super:
{
  # virtualised hosts don't work with `-cpu max`
  vmTools = super.vmTools // {qemu-common = import ./qemu_common_patched.nix {lib = self.lib; pkgs = self.pkgs;};
}
```

**Comment by jbedo** [comment]
I think we should keep this open while it's a problem in nixpkgs. The work around of changing the cpu features is clunky as vmTools doesn't expose qemu-common. At the least we could rework vmTools to allow overriding. We should do something before the freeze (#140168) as this issue has broad consequences.

**Comment by mschwaig** [comment]
I ran into (I'm assuming) the same issue.

In my case what triggered it was `/dev/kvm` not being world read-writable, not by nested virtualization.

For now a change to the `-cpu` flag got merged to work around the issue and will also be backported to `21.11` and `21.05`.

I was able to verify that `qemu 6.2.0` will fix this properly in my case using the open PR for `rc1` (#146526).

Maybe someone who ran into this because of nested virtualization can verify that my assumption is correct and qemu 6.2 fixes it in that case as well (even without the already merged change to `-cpu max`).
